packages/eslint-plugin-react-hooks/src/index.ts

```diff
@@ -7,20 +7,23 @@
 import type {ESLint, Linter, Rule} from 'eslint';
 
 import ExhaustiveDeps from './rules/ExhaustiveDeps';
-import ReactCompiler from './rules/ReactCompiler';
+import {allRules, recommendedRules} from './shared/ReactCompiler';
 import RulesOfHooks from './rules/RulesOfHooks';
 
 // All rules
 const rules = {
   'exhaustive-deps': ExhaustiveDeps,
-  'react-compiler': ReactCompiler,
   'rules-of-hooks': RulesOfHooks,
+  ...allRules,
 } satisfies Record<string, Rule.RuleModule>;
 
 // Config rules
 const configRules = {
   'react-hooks/rules-of-hooks': 'error',
   'react-hooks/exhaustive-deps': 'warn',
+  ...Object.fromEntries(
+    Object.keys(recommendedRules).map(name => ['react-hooks/' + name, 'error']),
+  ),
```

Comment:
but by default only the recommended rules are activated. and this is all driven dynamically based on the categories